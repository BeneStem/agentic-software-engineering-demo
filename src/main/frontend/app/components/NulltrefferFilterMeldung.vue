<template>
  <div class="erk_finden__nulltreffermeldung">
    <div class="erk_finden__nulltreffermeldung__vogelkaefig">
      <B2kIcon name="VogelAnVase" class="erk_finden__nulltreffermeldung__vogel" />
      <TrennlinieDick class="erk_finden__nulltreffermeldung__trennlinie" />
    </div>
    <div class="erk_finden__nulltreffermeldung__ueberschrift">
      <UeberschriftDrei>Für diese Filterkombination haben wir leider keine Ergebnisse für dich. </UeberschriftDrei>
      <UeberschriftDrei>Versuchs noch mal!</UeberschriftDrei>
    </div>
    <Button
      typ="sekundaer"
      groesse="auto"
      class="erk_finden__nulltreffermeldung__filterruecksetzenbutton"
      @click="filterruecksetzen()"
    >
      Alle Filter zurücksetzen
    </Button>
  </div>
</template>

<script setup lang="ts">
import TrennlinieDick from '@blume2000/design-system/dist/komponenten/atome/TrennlinieDick.vue';
import B2kIcon from '@blume2000/design-system/dist/komponenten/atome/B2kIcon.vue';
import UeberschriftDrei from '@blume2000/design-system/dist/komponenten/content/UeberschriftDrei.vue';
import Button from '@blume2000/design-system/dist/komponenten/interaktion/Button.vue';
import { useStore } from 'vuex';

const store = useStore();

const filterruecksetzen = () => {
  store.commit('ProduktfilterState/setAlleFilterZuruecksetzen');
};
</script>

<style lang="scss" scoped>
.erk_finden__nulltreffermeldung {
  margin-top: 72px;

  &__ueberschrift {
    margin: 24px 0;
  }

  &__vogelkaefig {
    position: relative;
    margin-top: 72px;
    margin-bottom: 24px;
  }

  &__vogel {
    position: absolute;
    top: -67px;
    height: 92px;
    width: 128px;
    --hauptfarbe: #{$color-dark};
  }

  &__trennlinie {
    --hauptfarbe: #{$color-dark};
  }
}

@media (min-width: $breakpoint-medium) {
  .erk_finden__nulltreffermeldung {
    margin: 128px 0;

    &__filterruecksetzenbutton {
      padding-top: 24px;
    }
  }
}
</style>
