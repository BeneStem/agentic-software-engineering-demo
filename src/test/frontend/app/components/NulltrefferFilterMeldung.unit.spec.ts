import { createStore } from 'vuex';
import { mount } from '@vue/test-utils';

import NulltrefferFilterMeldung from '@/main/frontend/app/components/NulltrefferFilterMeldung.vue';

describe('NulltrefferFilterMeldung', () => {
  it('renders correctly', () => {
    // Given
    const mockStateMutation = jest.fn();
    const store = createStore({
      mutations: {
        'ProduktfilterState/setAlleFilterZuruecksetzen': mockStateMutation
      }
    });
    const wrapper = mount(NulltrefferFilterMeldung, {
      global: {
        plugins: [store]
      }
    });

    // Then
    expect(wrapper.element).toMatchSnapshot();
  });

  it('triggers a store event on filter rÃ¼cksetzten click', () => {
    // Given
    const mockStateMutation = jest.fn();
    const store = createStore({
      mutations: {
        'ProduktfilterState/setAlleFilterZuruecksetzen': mockStateMutation
      }
    });

    const wrapper = mount(NulltrefferFilterMeldung, {
      global: {
        plugins: [store]
      }
    });

    // Then
    wrapper.find('.erk_finden__nulltreffermeldung__filterruecksetzenbutton').trigger('click');

    expect(mockStateMutation.mock.calls.length).toBe(1);
  });
});
