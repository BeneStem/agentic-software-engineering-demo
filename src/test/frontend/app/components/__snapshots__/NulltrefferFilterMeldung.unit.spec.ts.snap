// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NulltrefferFilterMeldung renders correctly 1`] = `
<div
  class="erk_finden__nulltreffermeldung"
>
  <div
    class="erk_finden__nulltreffermeldung__vogelkaefig"
  >
    <svg
      aria-hidden="true"
      class="erk_finden__nulltreffermeldung__vogel"
      role="img"
    >
      <!--v-if-->
      <use
        href="/static/assets/svgs/icons/VogelAnVase.svg#svg"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="sha_svg__trennlinie-dick erk_finden__nulltreffermeldung__trennlinie"
      role="img"
    >
      <!--v-if-->
      <use
        href="/static/assets/svgs/icons/TrennlinieDick.svg#svg"
      />
    </svg>
  </div>
  <div
    class="erk_finden__nulltreffermeldung__ueberschrift"
  >
    <h3
      class="sha_ueberschrift-drei sha_ueberschrift-drei--hell"
    >
      
      Für diese Filterkombination haben wir leider keine Ergebnisse für dich. 
      
    </h3>
    <h3
      class="sha_ueberschrift-drei sha_ueberschrift-drei--hell"
    >
      
      Versuchs noch mal!
      
    </h3>
  </div>
  <div
    class="sha_button sha_button--sekundaer-hell sha_button--auto erk_finden__nulltreffermeldung__filterruecksetzenbutton"
    role="button"
    tabindex="0"
  >
    <div
      class="sha_button__left"
    >
      <svg
        height="48"
        preserveAspectRatio="none"
        viewBox="0 0 22 44"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="activeGradient"
            x1="0%"
            x2="0%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#15a080"
            />
            <stop
              offset="30%"
              stop-color="#1fcca3"
            />
            <stop
              offset="100%"
              stop-color="#26d5ac"
            />
          </lineargradient>
        </defs>
        <path
          class="sha_button--background"
          d="M693.85,364.38v-.13c-1.78.05-13.08.24-14.82.58-2.89.77-5.07,3.05-5.61,7.42-.27,3-.42,6-.43,9a.67.67,0,0,0,0,.24v.2h0c0,1.42-.05,2.84-.08,4.26,0,.36-.06.72-.08,1.08a32,32,0,0,0,.44,11.66,7.76,7.76,0,0,0,2.46,3.78,8,8,0,0,0,6.64,1.92c1-.07,2.09-.08,3.13-.24,0,0,8.38,0,8.42-.07m-14.6-24c-.08-.16-.15-.32-.22-.48a1.45,1.45,0,0,1,.55,0c.12.05.14.23.2.35Zm.26,8.93-.6.22h0c-.07-.1-.24-.24-.2-.29a1.25,1.25,0,0,1,.45-.18ZM678.39,370a2.93,2.93,0,0,1,.74-.45h0a4,4,0,0,1,.75.55s-.28.33-.49.36c-.64.1-1.2-.19-1-.47M674,396.87l.15.84c0-.27-.1-.54-.15-.84m1.23-24.27c0,.44-.86-.43-.77,0a11.35,11.35,0,0,1,0,5.12c0,.16-.3.91-.59,1.72.1-1.56.19-3.13.25-4.71a10.14,10.14,0,0,1,2.17-6.48c-.18,1.41-.89,2.94-1.06,4.35m.19,28.06c.54.37,1.21.8,1.67,1.13a4.42,4.42,0,0,1-1.67-1.13m2.31-29.06a4.13,4.13,0,0,1-.8.59c-.27-.26-.58-.5-.53-.55h0a1.89,1.89,0,0,1,.82-.42c.1,0,.55.33.51.38m2,31c-.53-.13-1-.25-1.45-.38.73-.1,1.76-.3,2-.07a4.76,4.76,0,0,1,.57.62,8.56,8.56,0,0,1-1.13-.17m1.13-34.33a.86.86,0,0,1-.52-.37,4.57,4.57,0,0,1,.79-.6c.3.27.47.42.63.58a4.53,4.53,0,0,1-.9.39m.56,2.9c-.48-.39-.37-.65,0-.83a1.5,1.5,0,0,1,.52.09Zm1.14,9.8h0a.85.85,0,0,1-.16-.17,4,4,0,0,1,.49-.15.65.65,0,0,1,.18.15,3.81,3.81,0,0,1-.51.17m2.07-5.82h0a4.49,4.49,0,0,1-.75.47c-.14-.27-.22-.45-.3-.63a3.37,3.37,0,0,1,.78-.15c.1,0,.31.27.27.31m3.16,1.71a1.48,1.48,0,0,1-.54.25h0l-.66-.31a1.36,1.36,0,0,1,.48-.44c.35-.09.87.3.72.5m-.14-5.67h0a.87.87,0,0,1,1-.53c.14.07.18.23.34.47l-1,.53c-.13-.15-.4-.32-.36-.46m1.32,4.13h0a1.15,1.15,0,0,1-.44-.41,1.83,1.83,0,0,1,.67-.26,1.87,1.87,0,0,1,.62.45,2.49,2.49,0,0,1-.84.22m2.48-2.17-.77-.35.75-.56c.19.14.51.28.52.42s-.3.3-.5.49"
          transform="translate(-672.13 -362.1)"
        />
        <path
          class="sha_button--border"
          d="M680.9,403.67c-5.5-.95-6-3.48-6.7-7.15-.05-1.24-.07-2.42,0-3.66h-2a24.07,24.07,0,0,0,.82,8.23,7.09,7.09,0,0,0,3,3.28,12.45,12.45,0,0,0,7.27,1.73H694V404c-2,0-11.15,0-13.1-.36"
          transform="translate(-672.13 -362.1)"
        />
        <path
          class="sha_button--border"
          d="M674.45,375.74c.12-1.37-.07-2.78,0-4.15.33-3.7,1.88-5.62,5.32-6.21a22,22,0,0,1,5-.24h9.13V363l-7.8-.07c-2.13,0-4-.23-6.25,0-3.41.26-6.49,2.26-7.14,6.06-.18,1.36-.14,2.72-.23,4.1a14,14,0,0,0,0,1.45c0,.39,0,.79,0,1.18Z"
          transform="translate(-672.13 -362.1)"
        />
        <path
          class="sha_button--border"
          d="M672.48,376.87c-.11,1.41-.2,8.63-.26,10l-.07,1.79c-.06,1.44.08,2.72.07,4.16h2a40.24,40.24,0,0,0,.26-5.37c-.09-1.7.25-6.92,0-9,0,0,0-1.33,0-1.33a10.75,10.75,0,0,0,0-1.39l-2-.08s0,1.09,0,1.2"
          transform="translate(-672.13 -362.1)"
        />
      </svg>
    </div>
    <div
      class="sha_button__center"
    >
      <svg
        height="48"
        preserveAspectRatio="none"
        viewBox="0 0 80 44"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="activeGradient"
            x1="0%"
            x2="0%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#15a080"
            />
            <stop
              offset="30%"
              stop-color="#1fcca3"
            />
            <stop
              offset="100%"
              stop-color="#26d5ac"
            />
          </lineargradient>
        </defs>
        <path
          class="sha_button--background"
          d="M638.52,364.65h89v40h-89Z"
          transform="translate(-638.48 -362.51)"
        />
        <path
          class="sha_button--border"
          d="M727.52,403.65c-4.87-.11-9.8-.2-14.66-.11-8.55.17-57.8.45-72.7.82a1.84,1.84,0,0,0-.22-.17c-.31,0-1.08,0-1.39,0l-.07.37.07,1.92c12.57,0,61.53,0,68.45,0v-.22c6.82,0,13.7,0,20.52,0Z"
          transform="translate(-638.48 -362.51)"
        />
        <path
          class="sha_button--border"
          d="M640.32,365.57c8.24.06,53.42.09,61.65.09v-.23c8.41,0,17.14,0,25.55-.06v-2.73c-6.24-.15-12.81-.17-19-.06-9.92.17-54.2.49-66.56.89a.1.1,0,0,0,0,0c-.93-.06-1.87,0-2.82,0h-.52v2.11a17.83,17.83,0,0,0,1.8.07"
          transform="translate(-638.48 -362.51)"
        />
      </svg>
    </div>
    <div
      class="sha_button__text"
    >
      
       Alle Filter zurücksetzen 
      
    </div>
    <div
      class="sha_button__right"
    >
      <svg
        height="48"
        preserveAspectRatio="none"
        viewBox="0 0 23 44"
        width="23"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="activeGradient"
            x1="0%"
            x2="0%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#15a080"
            />
            <stop
              offset="30%"
              stop-color="#1fcca3"
            />
            <stop
              offset="100%"
              stop-color="#26d5ac"
            />
          </lineargradient>
        </defs>
        <path
          class="sha_button--background"
          d="M671.73,404.72v.12c1.78,0,13.08-.24,14.82-.57,2.88-.77,5.06-3,5.61-7.43.27-3,.42-6,.43-9a.69.69,0,0,0,0-.25v-.2h0c0-1.42,0-2.84.08-4.26,0-.35.06-.71.08-1.07a32,32,0,0,0-.44-11.67,7.85,7.85,0,0,0-2.46-3.78,8,8,0,0,0-6.64-1.92c-1,.08-2.09.09-3.13.25,0,0-8.38,0-8.42.06m14.6,24c.07.16.15.32.22.47a1.45,1.45,0,0,1-.55,0c-.12,0-.14-.23-.2-.35Zm-.27-8.93.61-.22h0c.07.1.24.24.2.29a1.77,1.77,0,0,1-.45.18Zm1.28,19.06a3,3,0,0,1-.74.46h0a4.52,4.52,0,0,1-.75-.55s.28-.34.49-.37c.64-.1,1.2.19,1,.47m4.36-26.91q-.07-.44-.15-.84.08.39.15.84m-1.23,24.26c0-.43.86.43.77,0a11.31,11.31,0,0,1,0-5.11c0-.17.3-.91.59-1.73-.1,1.57-.19,3.14-.25,4.71a10.17,10.17,0,0,1-2.17,6.49c.18-1.41.89-3,1.06-4.36m-.19-28.05c-.54-.37-1.21-.81-1.67-1.14a4.56,4.56,0,0,1,1.67,1.14m-2.31,29.05a4.59,4.59,0,0,1,.8-.59c.27.27.58.51.53.55h0a1.9,1.9,0,0,1-.82.43c-.1,0-.55-.33-.51-.38m-2-31c.53.13,1,.25,1.45.38-.73.1-1.76.3-2,.07a4.27,4.27,0,0,1-.57-.63,7.19,7.19,0,0,1,1.13.18m-1.13,34.32a.87.87,0,0,1,.52.38,4.57,4.57,0,0,1-.79.6l-.63-.58a3.61,3.61,0,0,1,.9-.39m-.56-2.9c.48.39.37.65,0,.83a1.5,1.5,0,0,1-.52-.09Zm-1.14-9.8h0a.65.65,0,0,1,.16.17,2.75,2.75,0,0,1-.49.14.53.53,0,0,1-.19-.15l.52-.16M680.89,394h0a4.49,4.49,0,0,1,.75-.47c.13.27.22.45.3.63a4.89,4.89,0,0,1-.78.15c-.1,0-.31-.28-.27-.31m-3.16-1.72a1.91,1.91,0,0,1,.54-.25h0l.66.31a1.3,1.3,0,0,1-.48.45c-.35.08-.87-.3-.72-.5m.14,5.67h0a.86.86,0,0,1-1,.52c-.14-.06-.18-.23-.34-.47l1-.53c.13.16.4.33.36.47m-1.32-4.14h0a1.18,1.18,0,0,1,.44.42,1.82,1.82,0,0,1-.67.25,1.71,1.71,0,0,1-.62-.44,2.76,2.76,0,0,1,.84-.23m-2.48,2.17.77.36-.75.55c-.19-.14-.51-.27-.52-.41s.3-.3.5-.5"
          transform="translate(-671.55 -362.61)"
        />
        <path
          class="sha_button--border"
          d="M692.22,374.43c0,.28,0,1.44,0,1.73h1.94v-1.41c-.1-.47.1-1-.09-1.43a14.22,14.22,0,0,0-.07-1.73c-.23-4.42-3.78-7.31-8.14-8.33a45.5,45.5,0,0,0-6.7-.64c-.16,0-7.45,0-7.61,0v2.77c2,0,9.93-.11,11.53.11,5.71.76,8.87,2.8,9.14,8.73"
          transform="translate(-671.55 -362.61)"
        />
        <path
          class="sha_button--border"
          d="M694,393.12h-2c0,.31,0,.3,0,.62-1.18.41.4,2.22,0,3.29-4,2.51.3,6.51-6.95,6.58a23.13,23.13,0,0,0-2.76.07c-.3-.05-1.23,0-1.52-.07-.87,0-8.12.22-9,.22v2.5h8.46c.62,0,1.24.08,1.85,0,1.3,0,2.66,0,3.8,0,3.48-.07,5-.68,7-3.41a13.5,13.5,0,0,0,1.16-5.34,5.13,5.13,0,0,1,0-1,9.32,9.32,0,0,0,0-1.83C694,394.6,693.94,393.58,694,393.12Z"
          transform="translate(-671.55 -362.61)"
        />
        <path
          class="sha_button--border"
          d="M692.19,377.28c-.12,1.4-.2,8.55-.26,10l-.07,1.77c-.06,1.43.07,2.69.07,4.12l2,.06a42,42,0,0,0,.26-5.38c-.08-1.68.26-6.85,0-8.94,0,0,0-1.32,0-1.32a12.71,12.71,0,0,0,0-1.38l-1.92-.1s0,1.11-.06,1.22"
          transform="translate(-671.55 -362.61)"
        />
      </svg>
    </div>
  </div>
</div>
`;
